src/main/java/core/basesyntax/strategy/impl/PurchaseOperation.java

```diff
@@ -0,0 +1,20 @@
+package core.basesyntax.strategy.impl;
+
+import core.basesyntax.db.Storage;
+import core.basesyntax.model.FruitTransaction;
+import core.basesyntax.strategy.OperationHandler;
+import java.util.Map;
+
+public class PurchaseOperation implements OperationHandler {
+    @Override
+    public void apply(FruitTransaction transaction) {
+        Map<String, Integer> fruits = Storage.getFruits();
+        String fruit = transaction.getFruit();
+        int quantity = transaction.getQuantity();
+        int currentQuantity = fruits.getOrDefault(fruit, 0);
+        if (currentQuantity < quantity) {
+            throw new RuntimeException("Not enough " + fruit + " in stock");
```

Comment:
According to the checklist, before saving the result balance in the Storage, you should check if the result balance is positive. The current check only ensures there is enough stock, but it does not prevent the balance from becoming negative. Ensure that the resulting balance is not negative before proceeding with the operation.